### PR TITLE
Allow deleting of a component group. Closes #286

### LINKED ISSUE #286: [Feature Request] Component group deletion
It would be nice to be able to delete component groups.

I'd say, if and only if there is no component associated with a group. However, if a component can be group-less, we could delete any component group, and make its components group-less.

### PR DESCRIPTION
**Pull request by jbrooksuk**
Simple `GET` request to delete.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
This is bad.

**Comment by GrahamCampbell** [comment]
GET must not have an "action". This violates the http spec. We must use `DELETE`.

**Comment by GrahamCampbell** [comment]
http://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html

**Comment by jbrooksuk** [comment]
Yeah, we need to tidy up the routes, components and incidents are deleted with a `GET` request too.

**Comment by GrahamCampbell** [comment]
Yeh, I just noticed. Very bad. Note that laravel support pretend DELETE requests.

**Comment by jbrooksuk** [comment]
I didn't know that?

**Comment by jbrooksuk** [comment]
To be honest, the whole `dashboard.php` routes file needs cleaning up.

**Comment by GrahamCampbell** [comment]
> I didn't know that?

One sec, I'll find the code, then explain what it means.

**Comment by GrahamCampbell** [comment]
Ok then, so laravel internally calls `Request::enableHttpMethodParameterOverride();` in https://github.com/laravel/framework/blob/v4.2.16/src/Illuminate/Foundation/start.php#L195 which is this: https://github.com/symfony/symfony/blob/v2.5.8/src/Symfony/Component/HttpFoundation/Request.php#L680.

This means you can send a POST request, and set the `_method` to `DELETE` and we're good.

**Comment by GrahamCampbell** [comment]
See the internals of how the method is determined by symfony: https://github.com/symfony/symfony/blob/v2.5.8/src/Symfony/Component/HttpFoundation/Request.php#L1246.

**Comment by GrahamCampbell** [comment]
I guess this will do for now. We can fix that issue later.

**Comment by GrahamCampbell** [comment]
This needs rebasing.

**Comment by jbrooksuk** [comment]
Done. @GrahamCampbell I'll take a proper look at Laravel faking these `DELETE` requests.

**Comment by GrahamCampbell** [comment]
I actually just use a small bit of javascript to do this in my projects.

**Comment by GrahamCampbell** [comment]
Feel free to lift the code out of my cms. :)

**Comment by jbrooksuk** [comment]
Sweet, thanks. I'll have a look now.

**Comment by GrahamCampbell** [comment]
https://github.com/BootstrapCMS/CMS/blob/v0.8.2-alpha/app/assets/js/cms-restfulizer.js
https://github.com/BootstrapCMS/CMS/blob/v0.8.2-alpha/app/views/events/delete.blade.php#L13

**Comment by jbrooksuk** [comment]
Looks easy enough :)

**Comment by jbrooksuk** [comment]
@GrahamCampbell mocked! Your code is basically unchanged too. Once this is merged, I'll make all of the delete routes work the same.

File: app/views/dashboard/components/groups/index.blade.php [hunk 1]
```diff
@@ -24,7 +24,7 @@
                         </div>
                         <div class='col-md-4 text-right'>
                             <a href='#' class='btn btn-default'>{{ trans('forms.edit') }}</a>
```
Comment (GrahamCampbell) [inline app/views/dashboard/components/groups/index.blade.php]: should we not just remove this button?

Comment (jbrooksuk) [inline app/views/dashboard/components/groups/index.blade.php]: Yeah we should for now.

**Comment by GrahamCampbell** [comment]
> Once this is merged, I'll make all of the delete routes work the same.

Might be better to just do it here, and rename this pull.

**Comment by jbrooksuk** [comment]
Oh, ok.

**Comment by jbrooksuk** [comment]
Done.

**Comment by jbrooksuk** [comment]
@GrahamCampbell all done. Anything else before I merge?

**Comment by GrahamCampbell** [comment]
Looks good. :)

**Comment by jbrooksuk** [comment]
Whoop.

**Pull request merged**